includes/Media.php

```diff
@@ -273,4 +295,31 @@ public static function upload_mimes( array $mime_types ) {
 		$mime_types['svg'] = 'image/svg+xml';
 		return $mime_types;
 	}
+
+	/**
+	 * Deletes associated poster image when a video is deleted.
+	 *
+	 * This prevents the poster image from becoming an orphan because it is not
+	 * displayed anywhere in WordPress or the story editor.
+	 *
+	 * @param int $attachment_id ID of the attachment to be deleted.
+	 *
+	 * @return void
+	 */
+	public static function delete_video_poster( $attachment_id ) {
+		$post_id = get_post_meta( $attachment_id, self::POSTER_ID_POST_META_KEY, true );
+		if ( empty( $post_id ) ) {
+			return;
+		}
+
+		if ( ! wp_attachment_is_image( $post_id ) ) {
```

Comment:
Is this really needed? 

We only set the `POSTER_POST_META_KEY` meta key on images anyway. So this seems redundant.

Reply:
It was more to check to see if the attachment existed before deleting. I was being overly careful here.